src/ServiceControl.Monitoring/Http/Diagrams/IntervalsAggregator.cs

```diff
@@ -37,11 +37,11 @@ internal static MonitoredEndpointValues AggregateRetries(List<IntervalsStore.End
             };
         }
 
-        internal static MonitoredEndpointValues AggregateQueueLength(List<IntervalsStore.EndpointInstanceIntervals> intervals)
+        internal static MonitoredValues AggregateQueueLength<T>(List<IntervalsStore<T>.IntervalsBreakdown> intervals, HistoryPeriod period)
```

Comment:
Just thinking now. We could use arrays instead of lists in all these methods, correct?

Reply:
Yes this is correct. That being this in worst case this collection holds as many items as there are instances, message-types per logical endpoint - so I'm not sure how much we can gain here. 

I propose to wait until we have performance tests for query side.